### PR TITLE
Merge pre-reg and at-door

### PR DESCRIPTION
**Pull request by kitsuta**
Depends on #1243.

This ports over the changes I made in Anthrocon to save attendees as they're registering - however, it's broken for groups. @EliAndrewC if you can spare the time, I'd like you to take a look at this and see what's wrong. Merging the group with the DB does something strange with the group leader.

### CODE REVIEW

**Comment by EliAndrewC** [comment]
I'll take a look this weekend after we have Dealer reg launched and whatnot :)

**Comment by binary1230** [comment]
FYI, I just merged latest master into this branch in order to make the pull request easier to read

**Comment by kitsuta** [comment]
Oh, also, this fixes #1042.

**Comment by binary1230** [comment]
hey because I'm dense and/or wasn't paying attention, don't let it hold up the PR but, what does "merge pre-reg and at-door" mean? am I reading this right that it adds the at-door payment options to the prereg page, so folks registering on their phones or whatever can select 'pay now' or 'pay later' etc?

**Comment by kitsuta** [comment]
Yup, that's one of the changes.

It effectively attempts to render `/registration/register` obsolete - we would always use the `/preregistration/form` page on-site.

One of the side-effects of this is to start saving people's registrations before they complete payment. Combined with the new badge statuses, we can now automatically detect if someone starts registering but doesn't finish, and bug them appropriately.

This worked great in Anthrocon, but Anthrocon never used groups, which are tripping me up. I made the PR before it was complete so Eli could get a change to review it and see if there's anything obvious that I'm doing wrong.

**Comment by EliAndrewC** [comment]
I'd planned on looking at this over the weekend, but fixing the unit tests took way longer than I'd hoped.

I'm somewhat concerned about people creating unpaid groups, since that creates a LOT of unused badges in the system.  Then again, if we have an automated process for cleaning them up, it's probably not as big a deal.

Regardless, I'll try to take some time to look at this during this week.

**Comment by earl7399** [comment]
Hmmm...another reason to get rid of groups.  Hmmm....

--
Robert Earl <r.a.earl@gmail.com>
FAX: 1-440-919-5086

**Comment by binary1230** [comment]
quick Q: are we still trying to merge this before classic? just thinking about if I should keep it on my radar or not

**Comment by EliAndrewC** [comment]
> quick Q: are we still trying to merge this before classic? just thinking about if I should keep it on my radar or not

If it's something that @kitsuta wants to do for MAGFest 2016 then we should probably make sure this gets merged and tested before MagClassic, since that gives us an opportunity to try it out at a smaller event to work out the kinks and such.

I've been losing a lot of sleep over the past month just trying to keep up with all of the other stuff that's been coming in, so I just haven't had any spare time to devote to this, so maybe it won't get done by Classic, but we'll see whether things settle down over the next couple of weeks.

**Comment by kitsuta** [comment]
Yeah, this is still a goal of mine, but we're not keeping up with the other issues.

File: uber/site_sections/preregistration.py [hunk 1]
```diff
@@ -267,9 +285,16 @@ def paid_preregistrations(self, session, payment_received=None):
                 'total_cost': payment_received
             }
 
-    def delete(self, id):
+    def delete(self, session, id):
+        try:
+            attendee = session.attendee(id)
+            attendee.badge_status = c.INVALID_STATUS
+            session.merge(attendee)
```
Comment (binary1230) [inline uber/site_sections/preregistration.py]: Redundant. --Eli/Vicki

**Comment by EliAndrewC** [comment]
I'm closing as part of old PR cleanup this because it's a year and a half old and needs a lot of work and will be somewhat OBE by the multi-event system we're designing.  With that being said, as we work on the design of the multi-event system, one of the things we'll need to decide is whether to store unpaid preregistrations in the database or not, so we should revisit that question when it comes up.

I'm also leaving the branch so we can refer to the code that's been written here late if we want.

**Pull request closed without merging**